### PR TITLE
Add secret engine for MySQL

### PR DESCRIPTION
**Pull request by ltcarbonell**
Adds the MySQL secret engine benchmark tests

### CODE REVIEW

File: benchmarktests/target_secret_mysql.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+// Postgres Secret Test Struct
+type MySQLSecret struct {
+	pathPrefix string
+	roleName   string
+	header     http.Header
+	config     *MySQLSecretTestConfig
+	logger     hclog.Logger
+}
+
+// Intermediary struct to assist with HCL decoding
```
Comment (ltcarbonell) [inline benchmarktests/target_secret_mysql.go:45]:
```suggestion
// Main Config Struct
```

File: docs/tests/secret-mysql.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+- `allowed_roles` `(list: ["benchmark_role"])` - List of the roles allowed to use this connection.
+- `root_rotation_statements` `(list: [])` - Specifies the database statements to be executed to rotate the root user's credentials.
+- `password_policy` `(string: "")` - The name of the password policy to use when generating passwords for this database. If not specified, this will use a default policy defined as: 20 characters with at least 1 uppercase, 1 lowercase, 1 number, and 1 dash character.
+- `connection_url` `(string: <required>)` - Specifies the MySQL DSN. This field can be templated and supports passing the username and password parameters in the following format `{{field_name}}`. A templated connection URL is required when using root credential rotation.
+- `disable_escaping` `(boolean: false)` - Turns off the escaping of special characters inside of the username and password fields. See the [databases secrets engine docs](https://developer.hashicorp.com/vault/docs/secrets/databases#disable-character-escaping) for more information. Defaults to `false`.
+- `max_open_connections` `(int: 4)` - Specifies the maximum number of open connections to the database.
```
Comment (elliesterner) [inline docs/tests/secret-mysql.md:18]: just out of curiosity have we tested this? if we don't set `max_open_connections` will it get sent to the api as the zero value (0)?

File: benchmarktests/target_secret_mysql.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+}
+
+// MySQL Role Config
+type MySQLRoleConfig struct {
+	Name                 string `hcl:"name,optional"`
+	DBName               string `hcl:"db_name,optional"`
+	DefaultTTL           string `hcl:"default_ttl,optional"`
+	MaxTTL               string `hcl:"max_ttl,optional"`
+	CreationStatements   string `hcl:"creation_statements"`
+	RevocationStatements string `hcl:"revocation_statements,optional"`
```
Comment (elliesterner) [inline benchmarktests/target_secret_mysql.go:81]: in the docs this is a slice

Comment (ltcarbonell) [reply benchmarktests/target_secret_mysql.go:81]: Ill leave it as a string and update the docs.  Good catch

File: docs/tests/secret-mysql.md (hunk 2 above)
Comment (ltcarbonell) [reply docs/tests/secret-mysql.md:18]: No it doesn't look like it.  Turns out even if we _explicitly_ set it to 0 (not sure why someone would, feel like that would defeat the purpose) it gets set to 4 also https://github.com/hashicorp/vault/blob/476bec104e2519b9d3454669cda6eba012a8a44c/plugins/database/mysql/connection_producer.go#L78-L79.

File: benchmarktests/target_secret_mysql.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+				Username:     os.Getenv(MySQLUsernameEnvVar),
+				Password:     os.Getenv(MySQLPasswordEnvVar),
+			},
+			MySQLRoleConfig: &MySQLRoleConfig{
+				Name:   "benchmark-role",
+				DBName: "benchmark-mysql",
+			},
+		},
+	}
+
```
Comment (jasonodonnell) [inline benchmarktests/target_secret_mysql.go:106]: Looks like an error is missing if both the secret envs and the config are nil?

**Pull request merged**